### DISCUSSION TITLE
Hydration Error on Random Content

### DISCUSSION THREAD

**Discussion by Geometrically** [nuxt/nuxt#18530] [Questions] [answered]
Hello!

I have a page with random content from an API route. However, when hydrating, the content is differently understandably so it causes a hydration error. How should this be prevented? Should I just ignore it? Here's an example which causes it:

```js
<script setup>
const number = ref(Math.random())
</script>

<template>
  <div>
    {{ number }}
  </div>
</template>
``` 

Thank you!

**Comment by TheAlexLichter** [comment, marked as answer]
Because the setup function will be executed on both, server- and client-side, the contents are different and there will be a hydration error.

*Do not ignore it*. Instead, you can use `useState` to "transfer" the value from server to client-side.
See also more in https://github.com/nuxt/nuxt/discussions/16828
